magda-web-client/src/Components/RequestDataset/RequestFormLogic.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import RequestFormTemplate from "./RequestFormTemplate";
+import Alert from "./Alert";
+import { correspondenceApiUrl, correspondenceApiReportUrl } from "../../config";
+
+export default class RequestFormLogic extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            successResult: false,
+            posted: false
+        };
+    }
+
+    /**
+     * handles the logic of submitting form
+     * @param {Object} data form submitted data from child component "RequestFormTemplate.js"
+     */
+    handleSubmit = data => {
+        const senderEmail = data.senderEmail;
+        const message = data.message;
+        const senderName = data.senderName;
+        var url = "";
+        //check to see the type of request. This will change the api url accordingly
+        if (this.props.requestType === "request") {
+            url = correspondenceApiUrl;
+        } else {
+            url = correspondenceApiReportUrl.replace(
+                ":datasetId",
+                this.props.datasetId
+            );
+        }
+        fetch(url, {
+            method: "POST",
+            headers: {
+                "Content-type": "application/json",
+                "Access-Control-Allow-Origin": "*"
+            },
+            body: JSON.stringify({
+                message,
+                senderEmail,
+                senderName
+            })
+        }).then(response => {
```

Comment:
This promise needs a `catch` too in case of network errors... it should just do the same thing as the `if (!response.ok)` block